### ISSUE TITLE
Train from Scratch Multispeaker with LJS and Custom Dataset

### ISSUE THREAD

**Issue by ArEnSc** [ming024/FastSpeech2#56]
Hey I have a custom datasets, that contains about 12 hours of data, I want to pre-train with multi speaker using LJSSpeech. 

Would I take the pretrained FS2 - LJS Speech model
Go into model.yml

change to 
multi_speaker: True
 speaker:  'universal'

ensure that the preprocessor outputs my data like so ?
6694_70837_000012_000001|6694|{P IY1 S}|peace!
name of file | speaker | alignment | utterance?

**Comment by ming024** [comment]
@ArEnSc I think it's OK. One more thing you should notice is that the size of the speaker embedding table may be different so some errors may occur when you load in the pretrained model. But I think it is not difficult to solve it.

**Comment by ArEnSc** [comment]
oh so it would be size of 1 and I would just need modify it to be 2

**Comment by ArEnSc** [comment]
I also heard that I should pad my wav files with 1 second before and after when using Montreal F Aligner, can you confirm this  ?

**Comment by ArEnSc** [comment]
> @ArEnSc I think it's OK. One more thing you should notice is that the size of the speaker embedding table may be different so some errors may occur when you load in the pretrained model. But I think it is not difficult to solve it.

could you show me what in the code I would have to change? I am planning to optimize most of the work you have done here, so that if name of file | speaker | alignment | utterance? is provided a model can be trained instantly

**Comment by ArEnSc** [comment]
Okay I think I found it

```
class FastSpeech2(nn.Module):
    """ FastSpeech2 """

    def __init__(self, preprocess_config, model_config):
        super(FastSpeech2, self).__init__()
        self.model_config = model_config

        self.encoder = Encoder(model_config)
        self.variance_adaptor = VarianceAdaptor(preprocess_config, model_config)
        self.decoder = Decoder(model_config)
        self.mel_linear = nn.Linear(
            model_config["transformer"]["decoder_hidden"],
            preprocess_config["preprocessing"]["mel"]["n_mel_channels"],
        )
        self.postnet = PostNet()

        self.speaker_emb = None
        if model_config["multi_speaker"]:
            with open(
                os.path.join(
                    preprocess_config["path"]["preprocessed_path"], "speakers.json"
                ),
                "r",
            ) as f:
                n_speaker = len(json.load(f))
Write Preview
 
Attach files by dragging & dropping, selecting or pasting them.
Add single comment
Cancel
            self.speaker_emb = nn.Embedding(
                n_speaker,
                model_config["transformer"]["encoder_hidden"],
            )

```

**Comment by ming024** [comment]
@ArEnSc You should load in the LJ-pretrained model first, and then set ``model.speaker_emb`` to be a newly-initialized embedding table before starting your multi-speaker training. As for the rumor about the MFA padding, I have to say I never heard and thought about it, maybe it works or maybe not. Just try it if you are interested!

**Comment by ming024** [comment]
closed #56
